transport/tls_no_pkcs11.go

```diff
@@ -0,0 +1,18 @@
+//go:build disable_pkcs11
+
+package transport
+
+import (
+	"crypto/tls"
+	"fmt"
+	"os"
+
+	"github.com/foundriesio/fioconfig/sotatoml"
+)
+
+
+func loadCertPkcs11(cfg *sotatoml.AppConfig) (*interface{}, tls.Certificate, error) {
+	fmt.Println("ERROR: PKCS#11 is not supported")
+	os.Exit(1)
```

Comment:
I'd rather panic here... so that it is clear that a code use is inappropriate.

See an example here: https://github.com/foundriesio/fioctl/blob/main/x509/bash.go#L162

Although as per your `GetTlsConfig` implementation, it looks like a user can erroneously get into this line via an incorrect config (when PKCS11 is disabled), right?
In this case your clean error makes sense.

Reply:
> Although as per your GetTlsConfig implementation, it looks like a user can erroneously get into this line via an incorrect config (when PKCS11 is disabled), right?

Yes, setting up the use of HSM in the config file would lead to this error if pkcs111 is not enabled in the binary. I.e., an user error.